ibis/expr/types.py

```diff
@@ -341,6 +331,9 @@ def parent(self):
     def resolve_name(self):
         return self.name
 
+    def has_resolved_name(self):
```

Comment:
Can you put a short docstring here? I know there's quite a few docstrings missing elsewhere in this module, but I think we should reverse that trend.